source/draw/RasterMonochrome.ooc

```diff
@@ -76,41 +76,45 @@ RasterMonochrome: class extends RasterPacked {
 	_resizeNearestNeighbour: static func (source, result: This) {
 		resultBuffer := result buffer pointer
 		sourceBuffer := source buffer pointer
-		for (row in 0 .. result size height) {
-			sourceRow := (source size height * row) / result size height
-			for (column in 0 .. result size width) {
-				sourceColumn := (source size width * column) / result size width
-				resultBuffer[column + result stride * row] = sourceBuffer[sourceColumn + source stride * sourceRow]
+		(resultWidth, resultHeight, resultStride) := (result size width, result size height, result stride)
+		(sourceWidth, sourceHeight, sourceStride) := (source size width, source size height, source stride)
+		for (row in 0 .. resultHeight) {
+			sourceRow := (sourceHeight * row) / resultHeight
+			for (column in 0 .. resultWidth) {
+				sourceColumn := (sourceWidth * column) / resultWidth
+				resultBuffer[column + resultStride * row] = sourceBuffer[sourceColumn + sourceStride * sourceRow]
```

Comment:
Also, I think indexing reads more clearly as `[column + row * stride]` rather than `[column + stride * row]`. Minor detail though, just consider it next time.